actor/sync_mailbox.go

```diff
@@ -0,0 +1,55 @@
+package actor
+
+import gocontext "context"
+
+// SyncMailbox is used to synchronously send data, and wait for it to process before returning.
+type SyncMailbox[T any] struct {
+	mbx Mailbox[*Callback[T]]
+}
+
+func NewSyncMailbox[T any](opts ...MailboxOption) *SyncMailbox[T] {
+	return &SyncMailbox[T]{
+		mbx: NewMailbox[*Callback[T]](opts...),
+	}
```

Comment:
since this is `sync` mailbox i would be better (performance wise) if this one is created with `opt.AsChan` with some small buffer. because that mailbox does not create goroutine and therefore has less operations per send/receive action.